From: Quinn Tran <quinn.tran@cavium.com>
Date: Fri, 15 Feb 2019 12:40:15 -0800
Subject: [PATCH] qla2xxx: allow irqbalance control in non-MQ mode
References: bsc#1128979
Patch-Mainline: never, SLE15 specific patch

For initiator mode, current code default to Blk-MQ to
control IRQ affinity.  This mode block irqbalance from
controlling IRQ setting.  This patch will see if Blk-MQ
is enabled or not before letting Blk-MQ control the IRQ
setting.

Signed-off-by: Quinn Tran <quinn.tran@cavium.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/qla2xxx/qla_isr.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/scsi/qla2xxx/qla_isr.c
+++ b/drivers/scsi/qla2xxx/qla_isr.c
@@ -3450,7 +3450,8 @@ qla24xx_enable_msix(struct qla_hw_data *
 		min_vecs++;
 	}
 
-	if (USER_CTRL_IRQ(ha) || !ha->mqiobase) {
+	if (USER_CTRL_IRQ(ha) || !ha->mqiobase ||
+	    !shost_use_blk_mq(vha->host)) {
 		/* user wants to control IRQ setting for target mode */
 		ret = pci_alloc_irq_vectors(ha->pdev, min_vecs,
 		    ha->msix_count, PCI_IRQ_MSIX);
